pkg/controller.v1/pytorch/pytorchjob_controller_test.go

```diff
@@ -164,32 +192,335 @@ var _ = Describe("PyTorchJob controller", func() {
+
+			By("Checking the PyTorchJob has resumed conditions")
+			Eventually(func() []kubeflowv1.JobCondition {
+				Expect(testK8sClient.Get(ctx, jobKey, created)).Should(Succeed())
+				return created.Status.Conditions
+			}, testutil.Timeout, testutil.Interval).Should(BeComparableTo([]kubeflowv1.JobCondition{
+				{
+					Type:    kubeflowv1.JobCreated,
+					Status:  corev1.ConditionTrue,
+					Reason:  commonutil.NewReason(kubeflowv1.PytorchJobKind, commonutil.JobCreatedReason),
+					Message: fmt.Sprintf("PyTorchJob %s is created.", name),
+				},
+				{
+					Type:    kubeflowv1.JobSuspended,
+					Status:  corev1.ConditionFalse,
+					Reason:  commonutil.NewReason(kubeflowv1.PytorchJobKind, commonutil.JobResumedReason),
+					Message: fmt.Sprintf("PyTorchJob %s is resumed.", name),
+				},
+				{
+					Type:    kubeflowv1.JobRunning,
+					Status:  corev1.ConditionTrue,
+					Reason:  commonutil.NewReason(kubeflowv1.PytorchJobKind, commonutil.JobRunningReason),
+					Message: fmt.Sprintf("PyTorchJob %s is running.", name),
+				},
+			}, testutil.IgnoreJobConditionsTimes))
+		})
+
+		It("Should delete resources after PyTorchJob is suspended; Should resume PyTorchJob after PyTorchJob is unsuspended", func() {
+			By("By creating a new PyTorchJob")
+			job.Spec.PyTorchReplicaSpecs[kubeflowv1.PyTorchJobReplicaTypeWorker].Replicas = pointer.Int32(1)
+			Expect(testK8sClient.Create(ctx, job)).Should(Succeed())
+
+			created := &kubeflowv1.PyTorchJob{}
+			masterPod := &corev1.Pod{}
+			workerPod := &corev1.Pod{}
+			masterSvc := &corev1.Service{}
+			workerSvc := &corev1.Service{}
+
+			// We'll need to retry getting this newly created PyTorchJob, given that creation may not immediately happen.
+			By("Checking created PyTorchJob")
+			Eventually(func() bool {
+				err := testK8sClient.Get(ctx, jobKey, created)
+				return err == nil
+			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
+
+			var startTimeBeforeSuspended *metav1.Time
+			Eventually(func() *metav1.Time {
+				Expect(testK8sClient.Get(ctx, jobKey, created)).Should(Succeed())
+				startTimeBeforeSuspended = created.Status.StartTime
+				return startTimeBeforeSuspended
+			}, testutil.Timeout, testutil.Interval).ShouldNot(BeNil())
+
+			By("Checking the created pods and services")
+			Eventually(func() bool {
+				errMaster := testK8sClient.Get(ctx, masterKey, masterPod)
+				errWorker := testK8sClient.Get(ctx, worker0Key, workerPod)
+				return errMaster == nil && errWorker == nil
+			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
+			Eventually(func() bool {
+				errMaster := testK8sClient.Get(ctx, masterKey, masterSvc)
+				errWorker := testK8sClient.Get(ctx, worker0Key, workerSvc)
+				return errMaster == nil && errWorker == nil
+			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
+
+			By("Updating the pod's phase with Running")
+			masterPod.Status.Phase = corev1.PodRunning
+			masterPod.ResourceVersion = ""
```

Comment:
I'm wondering if this is needed. Maybe we can drop it?

Reply:
Same reason in the above.

Reply:
Done.